cps/src/main/java/org/jenkinsci/plugins/workflow/cps/EnvActionImpl.java

```diff
@@ -108,4 +109,23 @@
         owner = r;
     }
 
+    @Extension public static class Binder extends Singleton {
+        @Override public String getName() {
+            return "env";
+        }
+        @Override public Object getProperty(CpsScript script) throws Exception {
+            Run<?,?> run = script.$build();
+            if (run != null) {
+                EnvActionImpl action = run.getAction(EnvActionImpl.class);
+                if (action == null) {
+                    action = new EnvActionImpl();
+                    run.addAction(action);
+                }
+                return action;
```

Comment:
I'm saying that I think it would make sense to have one Vs having to get Actions etc (not intuitive). As I said, prob not relevant to this PR other than it highlighted that it was ugly (to me).